src/pgduckdb_types.cpp

```diff
@@ -695,113 +816,130 @@ numeric_typmod_scale(int32 typmod) {
 }
 
 duckdb::LogicalType
-ConvertPostgresToDuckColumnType(Form_pg_attribute &attribute) {
```

Comment:
Overall I like these general refactorings, but they are quite big and seem to have little to do with the actual decimal changes. I would have preferred if this was a separate PR, but I'm okay with keeping here this time.

Reply:
++ forked off there: https://github.com/duckdb/pg_duckdb/pull/473